packages/autoloader/src/AutoloadGenerator.php

```diff
@@ -0,0 +1,372 @@
+ */
+class AutoloadGenerator extends BaseGenerator {
+
+	private $devMode = false;
+	/**
+	 * @var IOInterface
+	 */
+	private $io;
+
+	/**
+	 * @var bool
+	 */
+	private $classMapAuthoritative = false;
+
+	/**
+	 * Instantiate a AutoloadGenerator object.
+	 *
+	 * @param IOInterface $io
+	 */
+	public function __construct( $io ) {
+		$this->io = $io;
+	}
+
+	/**
+	 * Whether or not generated autoloader considers the class map
+	 * authoritative.
+	 *
+	 * @param bool $classMapAuthoritative
+	 */
+	public function setClassMapAuthoritative( $classMapAuthoritative ) {
+		$this->classMapAuthoritative = (bool) $classMapAuthoritative;
+	}
+
+	/**
+	 * Dump the autoloader.
+	 *
+	 * @param Config                       $config
+	 * @param InstalledRepositoryInterface $localRepo
+	 * @param PackageInterface             $mainPackage
+	 * @param InstallationManager          $installationManager
+	 * @param string                       $targetDir
+	 * @param bool                         $scanPsr0Packages
+	 * @param string                       $suffix
+	 */
+	public function dump(
+		Config $config,
+		InstalledRepositoryInterface $localRepo,
+		PackageInterface $mainPackage,
+		InstallationManager $installationManager,
+		$targetDir,
+		$scanPsr0Packages = false,
+		$suffix = ''
+	) {
+
+		if ( $this->classMapAuthoritative ) {
+
+			// Force scanPsr0Packages when classmap is authoritative
+			$scanPsr0Packages = true;
+		}
+
+		$this->devMode = $scanPsr0Packages;
+
+		$filesystem = new Filesystem();
+		$filesystem->ensureDirectoryExists( $config->get( 'vendor-dir' ) );
+
+		$basePath   = $filesystem->normalizePath( realpath( getcwd() ) );
+		$vendorPath = $filesystem->normalizePath( realpath( $config->get( 'vendor-dir' ) ) );
+		$targetDir  = $vendorPath . '/' . $targetDir;
+		$filesystem->ensureDirectoryExists( $targetDir );
+
+		$vendorPathCode = $filesystem->findShortestPathCode( realpath( $targetDir ), $vendorPath, true );
+
+		$appBaseDirCode = $filesystem->findShortestPathCode( $vendorPath, $basePath, true );
+		$appBaseDirCode = str_replace( '__DIR__', '$vendorDir', $appBaseDirCode );
+
+		$packageMap = $this->buildPackageMap( $installationManager, $mainPackage, $localRepo->getCanonicalPackages() );
+		$autoloads  = $this->parseAutoloads( $packageMap, $mainPackage );
+
+		// add custom psr-0 autoloading if the root package has a target dir
+		$targetDirLoader = null;
+
+		$filesCode = '';
+		// TODO: Test out file autoloading
+		// $autoloads['files'] = new \RecursiveIteratorIterator(new \RecursiveArrayIterator($autoloads['files']));
+		// foreach ($autoloads['files'] as $functionFile) {
+		// don't include file if it is using PHP 5.3+ syntax
+		// https://bitbucket.org/xrstf/composer-php52/issue/4
+		// if ($this->isPHP53($functionFile)) {
+		// $filesCode .= '//       require ' . $this->getPathCode($filesystem, $basePath, $vendorPath,
+		// $functionFile) . "; // disabled because of PHP 5.3 syntax\n";
```

Comment:
We can clean up anything < 5.3 now - minimum PHP version in Jetpack is PHP >= 5.3.2.

Reply:
This ^^ let's get rid of all the pre-5.3 compat stuff.

Reply:
done